packages/layers/src/multiscale-image-layer/multiscale-image-layer.js

```diff
@@ -84,10 +83,40 @@ const MultiscaleImageLayer = class extends CompositeLayer {
       // The image-tile example works without, this but I have a feeling there is something
       // going on with our pyramids and/or rendering that is different.
       const resolution = Math.round(-z);
+
+      // Here we set up variables for checking whether or not we should cip the black border of incoming tiles
+      // at low resolutions i.e for zarr tiles.  We need to check a few things before trimming:
+      //  1. The height/width of the full image at the current resolution
+      //  produces an image smaller than the current tileSize
+      //  2. The incoming image is indeed padded out to the tile size
+      //  in some dimension i.e it should be clipped down to the smaller size
+      // Once these have been confirmed, we trim the tile by going over it in row major order,
+      // keeping only the data that is not out of the clipped bounds.
+      const planarSize = loader[0].shape.slice(-2);
```

Comment:
I don't think this is safe. What about RGB where the last dim is channel? `getImageSize`?

Reply:
Yes, good catch!